blobmodel/model.py

```diff
@@ -219,28 +222,33 @@ def _sum_up_blobs(
 
     def _compute_start_stop(self, blob: Blob, speed_up: bool, error: float):
         if speed_up:
-            _start = int(blob.t_init / self._geometry.dt)
             if blob.v_x == 0:
-                _stop = self._geometry.t.size
-            else:
-                # ignores t_drain when calculating stop time
-                _stop = np.minimum(
-                    self._geometry.t.size,
-                    _start
-                    + int(
-                        (
-                            -np.log(error * np.sqrt(np.pi))
-                            + self._geometry.Lx
-                            - blob.pos_x
-                        )
-                        / (blob.v_x * self._geometry.dt)
-                    ),
+                start = 0
+                stop = self._geometry.t.size
+                return start, stop
+            start = int(
+                (
+                    blob.t_init * blob.v_x
+                    + blob.width_prop * np.log(error * np.sqrt(np.pi))
+                    + blob.pos_x
                 )
-        else:
-            _start = 0
-            _stop = self._geometry.t.size
-
-        return _start, _stop
+                / (self._geometry.dt * blob.v_x)
+            )
+            # ignores t_drain when calculating stop time
+            stop = np.minimum(
+                self._geometry.t.size,
+                start
+                + int(
+                    (
+                        -blob.width_prop * np.log(error * np.sqrt(np.pi))
+                        + self._geometry.Lx
+                        - blob.pos_x
+                    )
+                    / (blob.v_x * self._geometry.dt)
+                ),
+            )
+            return start, stop
+        return 0, self._geometry.t.size
```

Comment:
this is equivalent to the  `blob.v_x==0` case. We could reduce a level of indentation by writing:
`if blob_v_x == 0 or not speed_up:`
`    return 0, self._geometry.t_size`
`speed_up code`
what do you think?